src/simulation/point.py

```diff
@@ -178,17 +181,35 @@ def process_advection(self, delta_time: float) -> None:
                 crossing_point = (round(x + i * advection_x / abs(advection_y)), y + i * advection_y / abs(advection_y))
             if self.engine.get_topography(crossing_point) == TopographyState.LAND:
                 next_x, next_y = map(int, crossing_point)
-                # to ensure that buffer will be zeroed in a next step, because oil can't go further
-                self.advection_buffer = np.array([next_x - x, next_y - y], dtype='float64')
+                advection_vector = [next_x - x, next_y - y]  # so it's not going through land in the next step
                 break
 
-        if 0 <= next_x < const.point_side_count and 0 <= next_y < const.point_side_count:
-            if (next_x, next_y) not in self.world:
-                self.world[(next_x, next_y)] = Point((next_x, next_y), self.initial_values, self.engine)
-                self.engine.points_changed.append((next_x, next_y))
-            self.world[(next_x, next_y)].oil_buffer.append((self.oil_mass, self.viscosity_dynamic, self.emulsification_rate))
-        self.advection_buffer -= np.array([next_x - x, next_y - y])
-        self.oil_mass = 0
+        fractional_part_x = advection_vector[0] % 1
+        fractional_part_y = advection_vector[1] % 1
+        if self.initial_values.neighbourhood == Neighbourhood.VON_NEUMANN:
```

Comment:
adv_buff = [1.1, 1.9]

x_mass = 10 * 0.1 / 2 -> 0.5
y_mass = 10 * 0.9 / 2 -> 4.5   

so after that: 
(1, 1) = 5    - 50%   (in moore -> 9%)
(2, 1) - 0.5  - 5%     (in moore -> 1%)
(1, 2) - 4.5  - 45%   (in moore -> 81%!)

Reply:
New implementation pushed